crates/miden-tx/src/tests/kernel_tests/test_fpi.rs

```diff
@@ -839,40 +879,339 @@ fn test_simple_nested_fpi() {
         .unwrap();
 }
 
+/// Test that code will panic in attempt to create more than 63 foreign accounts.
+///
+/// Attempt to create a 64th foreign account first triggers the assert during the account data
+/// loading, but we have an additional assert during the account stack push just in case.
 #[test]
 fn test_nested_fpi_stack_overflow() {
-    let mut foreign_accounts = Vec::new();
+    // use a custom thread to increase its stack capacity
+    std::thread::Builder::new().stack_size(10 * 1_048_576).spawn(||{
+        let mut foreign_accounts = Vec::new();
 
-    for foreign_account_index in 0..65 {
-        let foreign_account_code_source = format!("
-            use.miden::account
+        let last_foreign_account_code_source = "
+                use.miden::account
 
-            export.get_item_foreign_{foreign_account_index}
-                # make this foreign procedure unique to make sure that we invoke the procedure of the 
-                # foreign account, not the native one
-                push.1 drop
-                exec.account::get_item
+                export.get_item_foreign
+                    # make this foreign procedure unique to make sure that we invoke the procedure 
+                    # of the foreign account, not the native one
+                    push.1 drop
 
-                # return the first element of the resulting word
-                drop drop drop
-            end
-        ");
+                    # push the index of desired storage item
+                    push.0
+
+                    exec.account::get_item
+
+                    # return the first element of the resulting word
+                    drop drop drop
+
+                    # make sure that the resulting value equals 1
+                    assert
+                end
+        ";
 
-        let foreign_account_component = AccountComponent::compile(
-            foreign_account_code_source,
+        let storage_slots = vec![AccountStorage::mock_item_0().slot];
+        let last_foreign_account_component = AccountComponent::compile(
+            last_foreign_account_code_source,
             TransactionKernel::testing_assembler(),
-            vec![],
+            storage_slots,
         )
         .unwrap()
         .with_supports_all_types();
 
-        let foreign_account = AccountBuilder::new(ChaCha20Rng::from_entropy().gen())
-            .with_component(foreign_account_component)
+        let last_foreign_account = AccountBuilder::new(ChaCha20Rng::from_os_rng().random())
+            .with_component(last_foreign_account_component)
             .build_existing()
             .unwrap();
 
-        foreign_accounts.push(foreign_account)
-    }
+        foreign_accounts.push(last_foreign_account);
+
+        for foreign_account_index in 0..63 {
+            let next_account = foreign_accounts.last().unwrap();
```

Comment:
We're building a chain of accounts here like `[account_0, account_1, account_2]` so to me it would be more logical if we called this `prev_account`. With the above example, `next_account` would be `account_2` while we're building `account_3` so to me this is the "previous" one. What do you think?

Reply:
First I named it "previous_account", but than changed to the "next_account" because of the order we are calling them: to make this chain of accounts work I need to create them starting from the account which will be called last. 

So, for example, in case of three foreign accounts the stack would look like so: 
`[account_0, account_1, account_2, native_account]`. In terms of the stack order `account_1` will be previous related to the `account_2`, but in terms of the call order `account_2` will be called first and `account_1` will be called _next_. But if it looks confusing I can change the name back to the `previous_account`.

Reply:
I see. I guess it just depends on how you read it. Feel free to leave as-is :)